### ISSUE TITLE
Accessibility: Link text ambiguous [Featured Product]

### ISSUE THREAD

**Issue by vikrampm1** [woothemes/woocommerce#52138]
**Prerequisites (mark completed items with an [x]):**
- [x] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [x] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

### Issue Description:

This link does not include meaningful text in either the visible text, an aria-label, or hidden screen reader text. It is not clear to screen reader users that the link goes to a specific product page instead of the main shop. If multiple of these blocks are added on the same page, there's no way to differentiate them.

![Image](https://github.com/user-attachments/assets/05be419d-dbb5-45fb-b6a0-d759334640bd)

### Component or Page: 

Component: Featured Product Block     

### Name affected element:

Shop Now link

### Relevant code snippets:

`<div class="wp-block-button"><a class="wp-block-button__link wp-element-button" href="https://woo.equalizedigital.com/?product=aerodynamic-steel-table">Shop now</a></div>`

### Recommended fix:

Dynamically get the title of the product that is being linked and add it to the Shop Now link to provide additional context on the link.
 
 There are three possible fixes for this link:
 A) Change the visible link text to something meaningful when heard outside the surrounding context. ""Shop Product Name Now""
 B) Add an aria-label to make the link meaningful. aria-label=""Shop Now: Product Name""
 C) Add hidden screen reader text along with the visible link anchor to add context to the anchor. ""Shop now<span class=""sr-only"">: Product Name</span>""

Ensure that you keep the visible text at the beginning of the aria-label so it works for voice control.

**Comment by Manussakis** [comment]
This issue is similar to the #52136 one and can likely be closed for the same reason detailed in [this comment](https://github.com/woocommerce/woocommerce/issues/52136#issuecomment-2611218599).

cc: @vikrampm1

**Comment by MajoMaldonado** [comment]
@vikrampm1  I am testing issue WooCore-072 and found that the `aria-label` has not been implemented. Marking it as "Not Fixed."

cc: @amberhinds

**Comment by Manussakis** [comment]
@MajoMaldonado I believe this issue should be marked as "Verified Fixed" after [this comment](https://github.com/woocommerce/woocommerce/pull/55351#issuecomment-2872830033) from @gigitux 

cc: @amberhinds @vikrampm1

**Comment by MajoMaldonado** [comment]
@Manussakis Done.
